client/my-sites/themes/themes-selection.jsx

```diff
@@ -343,7 +343,13 @@ export const ConnectedThemesSelection = connect(
 		const themes = getThemesForQueryIgnoringPage( state, sourceSiteId, query ) || [];
 
 		const shouldFetchWpOrgThemes = forceWpOrgSearch && sourceSiteId !== 'wporg' && !! search; // Only fetch WP.org themes when searching a term.
-		const wpOrgQuery = { ...query, page: 1 }; // We limit the WP.org themes to one page only.
+		const wpOrgQuery = {
+			...query,
+			// We limit the WP.org themes to one page only.
+			page: 1,
+			// WP.com theme filters don't match WP.org ones, so we add them to the search term.
+			search: filter ? search + ' ' + filter.replaceAll( '+', ' ' ).replaceAll( '-', ' ' ) : search,
```

Comment:
You could do both replaces in one swoop by passing a regex like `/[+-]/g` rather than the symbol. Maybe not as obvious though.

Reply:
I think a regex would be more readable, and it's also a bit more efficient than searching the string twice.
```suggestion
			search: filter ? `${search} ${filter.replace(/[+-]/g, ' ')}` : search,
```

Reply:
Oh, regex, my old nemesis, we meet again! But I agree it looks more readable, so I changed it. Thanks!